src/index.ts

```diff
@@ -0,0 +1,376 @@
+// We need to import the augmented definitions "somewhere" in our project, however since we have
+// it in tsconfig as an override and the api/types has imports, it is not strictly required here.
+// Because of the tsconfig override, we could import from '@polkadot/{api, types}/augment'
+import './interfaces/augment-api';
+import './interfaces/augment-types';
+// import '@polkadot/api/augment';
+// import '@polkadot/types/augment';
+import protoType from '@fragcolor/protos';
+
+// all type stuff, the only one we are using here
+// import {AccountsInfo} from "sample-polkadotjs-typegen/interfaces/protos";
+
+// external imports
+import { ApiPromise, Keyring } from '@polkadot/api';
+
+// our local stuff
+import * as definitions from './interfaces/definitions';
+import { ProtosCategories } from "@polkadot/types/lookup";
+import { Categories, GetGenealogyParams, GetProtosParams } from './interfaces/protos';
+import { GetDefinitionsParams, GetInstanceOwnerParams, GetInstancesParams } from './interfaces/fragments';
+import { u16, Vec } from '@polkadot/types-codec';
+
+
+export type availableCategories = {
+    text?: 'plain' | 'json',
+    trait?: Vec<u16>, // or should this be Array<number>??
+    shards?: 'edn' | 'binary',
+    audio?: 'oggFile' | 'mp3File',
+    texture?: 'pngFile' | 'jpgFile',
+    vector?: 'svgFile' | 'ttfFile',
+    video?: 'mkvFile' | 'mp4File',
+    model?: 'gltfFile' | 'sdf' | 'physicsCollider',
+    binary?: 'wasmProgram' | 'wasmReactor' | 'blendFile',
+};
+
+export type getProtosFuncParams = {
+    desc: boolean, 
+    fromIndex: number, 
+    limit: number, 
+    metadatKeys: Array<string>, 
+    categories: Array<availableCategories>, 
+    available: boolean|null, 
+    tags: Array<string>, 
+    returnOwers: boolean, 
+    owner: string|null, 
+    excludeTags: Array<string>
+}
+
+export type protoUploadFuncParams = {
+    references: Array<string>, 
```

Comment:
actually could you check if it's possible to strongly type a u8 array of length 32? It might not be possible. Then I guess `string` is okay.

Reply:
But could you at least create a new type called `ProtoHash` or something and set it to `string` so that the user knows that ProtoHashes are expected.

(So instead of `Array<string>`, make it `Array<ProtoHash>`)

Reply:
^